eth-keystore/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: MIT
+
+//! Ethereum wallet key store.
+
+mod memory;
+mod persistent;
+
+use anyhow::Result;
+use std::hash::Hash;
+use zeroize::Zeroize;
+
+pub use crate::persistent::PersistentKeyStore;
+
+/// The key store trait for different evm key store
+pub trait KeyStore {
+    /// The type of the key that is stored
+    type Key: Clone + Eq + Hash + TryFrom<KeyInfo>;
+
+    /// Get the key info by address string
+    fn get(&self, addr: &Self::Key) -> Result<Option<KeyInfo>>;
+    /// List all addresses in the key store
+    fn list_all(&self) -> Result<Vec<Self::Key>>;
+    /// Put a new info to the addr
+    fn put(&mut self, info: KeyInfo) -> Result<()>;
+}
```

Comment:
We are missing a remove function. We will need it so I suggest including it already.
```suggestion
    fn remove(&mut self, addr: &Self::Key)
}
```